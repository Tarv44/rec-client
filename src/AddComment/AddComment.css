.song-form {
    display: flex;
    justify-content: space-between;
}

.comment-input {
    width: 100% !important;
    border: none;
    background-color: #acacac;
    color: white;
    border-radius: 4px;
    font-size: 16px;
    margin-right: 5px;
}

.comment-button {
    box-sizing: content-box;
}