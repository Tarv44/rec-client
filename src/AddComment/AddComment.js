import { Component } from "react";
import './AddComment.css';

export default class AddComment extends Component {
    render() {
        return (
            <form onSubmit={e => this.props.handleCommentSubmit(e, this.props.songIdx)}>
                <input 
                    type='text'
                    name='comment'
                    onChange={e => this.props.handleComment(e.target.value, this.props.songIdx)}
                />
                <button type='submit'>Comment</button>
            </form>
        )
    }
}