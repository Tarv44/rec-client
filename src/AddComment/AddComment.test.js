import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import AddComment from './AddComment'


describe('----- AddComment -----', () => {
    const handleComment = () => {console.log('comment submitted')}
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render(<AddComment/>, div)
        ReactDom.unmountComponentAtNode(div);
    })
    it('renders the UI as expected', () => {
        const comp = renderer
            .create(<AddComment
                handleComment={handleComment}
                songIdx={1}
                inputValue={'comment'}        
            />)
            .toJSON()
        expect(comp).toMatchSnapshot()
    })
})
