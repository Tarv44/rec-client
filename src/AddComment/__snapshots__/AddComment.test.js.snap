// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`----- AddComment ----- renders the UI as expected 1`] = `
<form
  autoComplete="off"
  className="song-form"
  onSubmit={[Function]}
>
  <input
    className="comment-input"
    maxLength="1500"
    name="comment"
    onChange={[Function]}
    type="text"
    value="comment"
  />
  <button
    className="comment-button"
    type="submit"
  >
    Comment
  </button>
</form>
`;
